src/main/java/seedu/address/model/person/Birthday.java

```diff
@@ -0,0 +1,119 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+
+/**
+ * Represents a Person's birthday in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidBirthday(String)}
+ */
+public class Birthday {
+
+    public static final String MESSAGE_BIRTHDAY_CONSTRAINTS =
+            "Person birthdays should be 6-8 numbers separated by '/', '-' or '.' with the format of day/month/year"
+            + "\nExamples of valid format: 01/12/1990, 6-3-1991, 18-07-1992";
+    public static final String BIRTHDAY_VALIDATION_REGEX = "^(0[1-9]|[12][\\d]|3[01]|[1-9])[///./-]"
+            + "(0[1-9]|1[012]|[1-9])[///./-](19|20)\\d\\d$";
+
+    public final String value;
+
+    /**
+     * Validates given birthday.
+     *
+     * @throws IllegalValueException if given birthday string is invalid.
+     */
+    public Birthday(String birthday) throws IllegalValueException {
+        requireNonNull(birthday);
+        if (birthday.equals("")) {
+            this.value = birthday;
+        } else {
+            String trimmedBirthday = birthday.trim();
+            if (!isValidBirthday(trimmedBirthday)) {
```

Comment:
Good job on the refactoring of the if statement. Try doing this for other if statements as well.